lua/xbase/treesitter.lua

```diff
@@ -0,0 +1,173 @@
+local have_treesitter = pcall(require, "nvim-treesitter")
+if not have_treesitter then
+    return nil
+end
+
+local ts_utils = require("nvim-treesitter.ts_utils")
+local get_node_text = function(node)
+	return vim.treesitter.get_node_text(node, vim.api.nvim_get_current_buf())
+end
+local get_winr = vim.api.nvim_get_current_win
+local get_current_node = ts_utils.get_node_at_cursor
+
+local property_declaration_query = vim.treesitter.query.parse("swift", "(property_declaration) @prop")
```

Comment:
I get this error:

treesitter.lua:13: attempt to call field 'parse' (a nil value), whenever I open nvim:

```
NVIM v0.8.3
Build type: Release
LuaJIT 2.1.0-beta3
Compiled by _nixbld11

```

Reply:
Ah yea, in 0.9 it changed from `vim.treesitter.parse_query` to `vim.treesitter.query.parse`. I can do a conditional check here based on the version.

Reply:
That's odd indeed.

Reply:
That works now, closing

Reply:
I just realized my nvim versioning was goofed, I reinstalled and I got the deprecation notice. I added in conditional logic to select the parser

Reply:
😅😅 Didn't see this one coming. Yeah I need to update mine soon as well